docs/docs/genai/concepts/scorers.mdx

```diff
@@ -0,0 +1,230 @@
+| ----------- | ---------------- | -------------------------------------------------------------------------- |
+| `name`      | `str`            | Unique identifier for the scorer (e.g., "correctness", "safety")           |
+| `value`     | `Any`            | The evaluation result - can be numeric, boolean, or categorical            |
+| `rationale` | `Optional[str]`  | Explanation of why this score was given (especially useful for LLM judges) |
+| `metadata`  | `Optional[dict]` | Additional information about the evaluation (confidence, sub-scores, etc.) |
+| `error`     | `Optional[str]`  | Error message if the scorer failed to evaluate                             |
+
+## Common Scorer Patterns
+
+### Binary Classification
+
+```python
+# Simple pass/fail evaluation
+def is_safe(outputs):
+    # Check for harmful content
+    return "harmful" not in outputs.lower()
+```
+
+### Numeric Scoring
+
+```python
+# Score on a scale (e.g., 0-1 or 1-5)
+def relevance_score(inputs, outputs):
+    # Calculate relevance score
+    return calculate_similarity(inputs, outputs)
+```
+
+### Categorical Rating
+
+```python
+# Multi-class evaluation
+def quality_rating(outputs):
+    # Classify into categories
+    if len(outputs) < 50:
+        return "poor"
+    elif len(outputs) < 200:
+        return "good"
+    else:
+        return "excellent"
+```
+
+### LLM Judge
+
+```python
+from mlflow.genai.judges import make_judge
+
+correctness_judge = make_judge(
+    name="correctness",
+    instructions=(
+        "Evaluate if the response in {{ outputs }} "
+        "correctly answers the question in {{ inputs }}."
+    ),
+    model="anthropic:/claude-3-opus-20240229",
+)
+```
+
+## Integration with MLflow Evaluation
+
+Scorers are the building blocks of MLflow's evaluation framework. They integrate seamlessly with `mlflow.genai.evaluate()`:
+
+```python
+import mlflow
+import pandas as pd
+
+# Your test data
+test_data = pd.DataFrame(
+    [
+        {
+            "question": "What is MLflow?",
+            "ground_truth": "MLflow is an open-source platform for managing the ML lifecycle",
+        },
+        {
+            "question": "How do I track experiments?",
+            "ground_truth": "Use mlflow.start_run() to track experiments",
+        },
+    ]
+)
+
+
+# Your application
+def my_app(question):
+    return generate_answer(question)
+
+
+# Evaluate with multiple scorers
+results = mlflow.genai.evaluate(
+    data=test_data,
+    predict_fn=my_app,
+    scorers=[
+        correctness_judge,  # LLM judge
+        relevance_score,  # Code-based scorer
+        is_safe,  # Binary classifier
+    ],
+)
+
+# Access evaluation metrics
+print(f"Correctness: {results.metrics['correctness/mean']}")
+print(f"Relevance: {results.metrics['relevance_score/mean']}")
+print(f"Safety Rate: {results.metrics['is_safe/ratio']}")
+```
+
+## Best Practices
+
+1. **Choose the Right Scorer Type**
+   - Use code-based scorers for objective, deterministic metrics
```

Comment:
Let's move code-based to the end of the list

Reply:
yep - refactored this whole section.